File: openapi-generator/dart-jaguar/lib/model/vehicle_maintenance_j1939.dart
import 'package:jaguar_serializer/jaguar_serializer.dart';

import 'package:openapi/model/vehicle_maintenance_j1939_check_engine_light.dart';
import 'package:openapi/model/vehicle_maintenance_j1939_diagnostic_trouble_codes.dart';
part 'vehicle_maintenance_j1939.jser.dart';

class VehicleMaintenanceJ1939 {
  
  @Alias('checkEngineLight')
  final VehicleMaintenanceJ1939CheckEngineLight checkEngineLight;
   /* J1939 DTCs. */
  @Alias('diagnosticTroubleCodes')
  final List<VehicleMaintenanceJ1939DiagnosticTroubleCodes> diagnosticTroubleCodes;
  

  VehicleMaintenanceJ1939(
      

{
     this.checkEngineLight = null,  
     this.diagnosticTroubleCodes = const [] 
    
    }
  );

  @override
  String toString() {
    return 'VehicleMaintenanceJ1939[checkEngineLight=$checkEngineLight, diagnosticTroubleCodes=$diagnosticTroubleCodes, ]';
  }
}

@GenSerializer(nullableFields: true)
class VehicleMaintenanceJ1939Serializer extends Serializer<VehicleMaintenanceJ1939> with _$VehicleMaintenanceJ1939Serializer {

}

File: openapi-generator/java-vertx/src/main/java/org/openapitools/server/api/model/VehicleMaintenanceJ1939.java
package org.openapitools.server.api.model;

import java.util.Objects;
import com.fasterxml.jackson.annotation.JsonInclude;
import com.fasterxml.jackson.annotation.JsonProperty;
import java.util.ArrayList;
import java.util.List;
import org.openapitools.server.api.model.VehicleMaintenanceJ1939CheckEngineLight;
import org.openapitools.server.api.model.VehicleMaintenanceJ1939DiagnosticTroubleCodes;

/**
 * J1939 based data. Null if no data is available.
 **/
@JsonInclude(JsonInclude.Include.NON_NULL) 
public class VehicleMaintenanceJ1939   {
  
  private VehicleMaintenanceJ1939CheckEngineLight checkEngineLight = null;
  private List<VehicleMaintenanceJ1939DiagnosticTroubleCodes> diagnosticTroubleCodes = new ArrayList<>();

  public VehicleMaintenanceJ1939 () {

  }

  public VehicleMaintenanceJ1939 (VehicleMaintenanceJ1939CheckEngineLight checkEngineLight, List<VehicleMaintenanceJ1939DiagnosticTroubleCodes> diagnosticTroubleCodes) {
    this.checkEngineLight = checkEngineLight;
    this.diagnosticTroubleCodes = diagnosticTroubleCodes;
  }

    
  @JsonProperty("checkEngineLight")
  public VehicleMaintenanceJ1939CheckEngineLight getCheckEngineLight() {
    return checkEngineLight;
  }
  public void setCheckEngineLight(VehicleMaintenanceJ1939CheckEngineLight checkEngineLight) {
    this.checkEngineLight = checkEngineLight;
  }

    
  @JsonProperty("diagnosticTroubleCodes")
  public List<VehicleMaintenanceJ1939DiagnosticTroubleCodes> getDiagnosticTroubleCodes() {
    return diagnosticTroubleCodes;
  }
  public void setDiagnosticTroubleCodes(List<VehicleMaintenanceJ1939DiagnosticTroubleCodes> diagnosticTroubleCodes) {
    this.diagnosticTroubleCodes = diagnosticTroubleCodes;
  }


  @Override
  public boolean equals(Object o) {
    if (this == o) {
      return true;
    }
    if (o == null || getClass() != o.getClass()) {
      return false;
    }
    VehicleMaintenanceJ1939 vehicleMaintenanceJ1939 = (VehicleMaintenanceJ1939) o;
    return Objects.equals(checkEngineLight, vehicleMaintenanceJ1939.checkEngineLight) &&
        Objects.equals(diagnosticTroubleCodes, vehicleMaintenanceJ1939.diagnosticTroubleCodes);
  }

  @Override
  public int hashCode() {
    return Objects.hash(checkEngineLight, diagnosticTroubleCodes);
  }

  @Override
  public String toString() {
    StringBuilder sb = new StringBuilder();
    sb.append("class VehicleMaintenanceJ1939 {\n");
    
    sb.append("    checkEngineLight: ").append(toIndentedString(checkEngineLight)).append("\n");
    sb.append("    diagnosticTroubleCodes: ").append(toIndentedString(diagnosticTroubleCodes)).append("\n");
    sb.append("}");
    return sb.toString();
  }

  /**
   * Convert the given object to string with each line indented by 4 spaces
   * (except the first line).
   */
  private String toIndentedString(Object o) {
    if (o == null) {
      return "null";
    }
    return o.toString().replace("\n", "\n    ");
  }
}
